File: app/src/main/java/de/openfiresource/falarm/models/database/OperationRuleDao.java
package de.openfiresource.falarm.models.database;


import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

@Dao
public interface OperationRuleDao {

    @Query("SELECT * FROM operation_rule")
    LiveData<List<OperationRule>> getAllAsync();

    @Query("SELECT * FROM operation_rule")
    List<OperationRule> getAll();

    @Query("SELECT * FROM operation_rule WHERE id = :id LIMIT 1")
    OperationRule findById(long id);

    @Query("SELECT * FROM operation_rule WHERE id = :id LIMIT 1")
    LiveData<OperationRule> findByIdAsync(long id);

    @Insert
    Long insertOperationRule(OperationRule operationRule);

    @Update
    void updateOperationRule(OperationRule operationRule);

    @Delete
    void deleteOperationRule(OperationRule operationRule);
}

File: app/src/main/java/de/openfiresource/openpager/models/database/OperationRuleDao.kt
package de.openfiresource.openpager.models.database


import android.arch.lifecycle.LiveData
import android.arch.persistence.room.*

@Dao
interface OperationRuleDao {

    @Query("SELECT * FROM operation_rule")
    fun allAsync(): LiveData<List<OperationRule>>

    @Query("SELECT * FROM operation_rule")
    fun all(): List<OperationRule>

    @Query("SELECT * FROM operation_rule WHERE id = :id LIMIT 1")
    fun findById(id: Long): OperationRule

    @Query("SELECT * FROM operation_rule WHERE id = :id LIMIT 1")
    fun findByIdAsync(id: Long): LiveData<OperationRule>

    @Insert
    fun insertOperationRule(operationRule: OperationRule): Long?

    @Update
    fun updateOperationRule(operationRule: OperationRule)

    @Delete
    fun deleteOperationRule(operationRule: OperationRule)
}
